### PR TITLE
vttablet: debug/env page to change variables in real-time

### LINKED ISSUE #2983: Feature: reconfigure components at run time without the need to restart them
This feature request is probably not very useful for container based deployments but I can see it pretty valuable for all the other setup.

### LINKED ISSUE #7171: RFC: Allow Real-time change of parameters
## Problem Statement

Vitess Components currently disallow changing of configuration parameters while a process is running. There are a few reasons why this was not allowed:

* The command line used to launch the process will cease to be an authoritative source.
* It is difficult to audit any values that were changed real-time, or keep track of history if changes were made multiple times.
* If a process is restarted for any reason, the changes will be reverted.

However, if there is an ongoing high severity incident, it may be beneficial to allow human operators to make temporary changes to a system. This, of course, comes with the caveat that they will eventually revert the changes or make them permanent by deploying the binaries with the new flags.

## Proposal

Given that this should be generally discouraged as common practice, these capabilities should not be made openly available. For example, you should not be able to make such changes using SQL statements.

We’ll export a `/debug/env` endpoint from vttablet, protected by `ADMIN` access. The URL will display the current values of the various parameters, and will allow you to modify and submit any of those.

Values like connection pool sizes, query timeouts, query consolidation settings, etc. will be changeable. We’ll iterate on this list as more use cases arise. The initial list will be in an upcoming PR.

## Other options considered

SET Statements: These would be issued via VTGate. The problem with this approach is that the commands would be too widely available. Also, VTGate doesn’t allow targeting of specific tablets.

vtctld ExecuteFetchAsDba SET statements: This could be made to work. However, this is currently implemented as pass-through in tablet manager. Significant changes will be needed to parse the statements and make them do other things.

vtcltd alternate command, like `SetTabletEnv`: This could be made to work. However, it is a lot more work than what is proposed, and may not be worth it for a feature that is meant to be used so rarely. We still have the option of implementing this if the need arises in the future.

### PR DESCRIPTION
**Pull request by sougou**
## Backport
NO

## Status
READY

## Description
Feature implemented as specified in RFC.
Also added links to the new page from `/debug/status`

## Related Issue(s)
Fixes #7171 
Fixes #2983

## Todos
- [X] Tests
- [ ] Documentation

## Deployment Notes
None

## Impacted Areas in Vitess
List general components of the application that this PR will affect:

- [X]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build

### CODE REVIEW

File: go/vt/vttablet/tabletserver/debugenv.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	</form></tr>
+	`))
+)
+
+type envValue struct {
+	VarName string
+	Value   string
+}
+
+func debugEnvHandler(tsv *TabletServer, w http.ResponseWriter, r *http.Request) {
```
Comment (rafael) [inline go/vt/vttablet/tabletserver/debugenv.go:56]: Probably we should reject the request if the method is not **POST**

**Review by rafael — APPROVED** [review]
@sougou this looks good to me.

File: go/vt/vttablet/tabletserver/debugenv.go (hunk 1 above)
Comment (sougou) [reply go/vt/vttablet/tabletserver/debugenv.go:56]: I actually rely on this for my tests, by using GET. Is there a reason to disallow? Also, we seem to be allowing it in other places.

Comment (rafael) [reply go/vt/vttablet/tabletserver/debugenv.go:56]: I saw that, If you change this, you will have to update the tests. 

I think the reason is general HTTP conventions. In theory, `GET` requests could be cached and shouldn't mutate state.

I think that ideally, we should stop propagating this pattern. Once people start relying on the get will become even more difficult to change it.

**Pull request merged**